### PR TITLE
alertmanager: add missing types

### LINKED ISSUE #5857: [alertmanager] webhook url_file parameter is missing
**What is missing?**

Since alertmanager v0.26.x is it possible to load an webhook url via `url_file`  https://github.com/prometheus/alertmanager/pull/3223

But it looks like the current prometheus-operator don't support it yet:

```
level=error ts=2023-08-28T15:42:18.393763099Z caller=klog.go:116 component=k8s_client_runtime func=ErrorDepth msg="sync \"monitoring-system/cluster\" failed: provision alertmanager configuration: failed to initialize from secret: yaml: unmarshal errors:\n  line 70: field url_file not found in type config.plain"
```

**Why do we need it?**

Allow a webhook url via secret

**Environment**

* Prometheus Operator version:

    `v0.67.1`
    
    

**Anything else we need to know?**:

If someone only have to add the new parameter 
```
URLFile string          `yaml:"url_file,omitempty" json:"url_file,omitempty"`
```
[here](https://github.com/prometheus-operator/prometheus-operator/blob/v0.67.1/pkg/alertmanager/types.go#L114)? 
I could also take over this. Just let me know

### PR DESCRIPTION
**Pull request by HaveFun83**
alertmanager v0.26 introduced some new types in receiver configuration

Fixes #5857

## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
add missing types introduced in alertmanager v0.26
```

### CODE REVIEW

**Review by simonpasquier — COMMENTED** [review]
Thanks for the PR! We need to sanitize the inputs based on the Alertmanager version. See an example here:
https://github.com/prometheus-operator/prometheus-operator/blob/fa1893b31b8046a25ba8795c554dd9c653765c12/pkg/alertmanager/amcfg.go#L1949-L1967

**Comment by HaveFun83** [comment]
> Thanks for the PR! We need to sanitize the inputs based on the Alertmanager version. See an example here:
> 
> https://github.com/prometheus-operator/prometheus-operator/blob/fa1893b31b8046a25ba8795c554dd9c653765c12/pkg/alertmanager/amcfg.go#L1949-L1967

Thank you for the hint i will try to implement it

File: pkg/alertmanager/amcfg.go [hunk 1]
```diff
@@ -1984,12 +2030,24 @@ func (tc *telegramConfig) sanitize(amVersion semver.Version, logger log.Logger)
...
+		tc.BotTokenFile = ""
+	}
+
 	if tc.ChatID == 0 {
 		return errors.Errorf("mandatory field %q is empty", "chatID")
 	}
 
-	if tc.BotToken == "" {
-		return fmt.Errorf("mandatory field %q is empty", "botToken")
+	if tc.BotToken == "" && amVersion.LT(semver.MustParse("0.26.0")) {
```
Comment (HaveFun83) [inline pkg/alertmanager/amcfg.go:2043]: maybe someone have a better idea to get this working with both parametes `botToken` or `botTokenFile`?

**Comment by slashpai** [comment]
For telegramConfigs I have added it in https://github.com/prometheus-operator/prometheus-operator/pull/5882

**Comment by HaveFun83** [comment]
> For telegramConfigs I have added it in #5882

ok than i remove the config from this PR.

**Comment by HaveFun83** [comment]
@simonpasquier @slashpai can someone have a look please?

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Pull request merged**